Gh-364: New getting started guide (#370)
gchq/gaffer-doc: docs/getting-started/example-deployment/using-the-api.md

# Using the API

!!! info "Work in Progress"

    This page is under construction.

As covered in the [Gaffer basics](../basics.md) the main POST request used in the API is
`/graph/operations/execute`. This part of the guide will cover the general usage of this part of the
API and walk through some general operations you might want to use to load data or query.

!!! note
    The [Gaffer operations reference guide](../../reference/operations-guide/operations.md)
    has more detail on available operations.

## Loading Data

Gaffer supports various aspects of loading data, generally this is done in a few stages which are

> outlined in the following diagram.

Reviewer: Loading data in production is not really done using the REST API with generate elements. There are some other loading methods used which have varying degrees of documentation or direct loading into Accumulo which bypasses Gaffer.
For this example it's enough to say e.g: for small amounts of data the following steps can be used.

# Using the API

!!! info "Work in Progress"

    This page is under construction.

As covered in the [Gaffer basics](../basics.md) the main POST request used in the API is
`/graph/operations/execute`. This part of the guide will cover the general usage of this part of the
API and walk through some general operations you might want to use to load data or query.

> !!! note
>     Please refer to the reference guide on [Gaffer
>     operations](../../reference/operations-guide/operations.md) for more detail on available
>     operations.

Reviewer: This is a case when forced line wrapping makes it harder to read I'd say.
```suggestion
!!! note
    The [Gaffer operations reference guide](../../reference/operations-guide/operations.md)
    has more detail on available operations.
```

                "class": "Neo4jCsvElementGenerator"
            }
        },
        {
            "class": "AddElements"
        }
    ]
}
```

The operation chain above essentially mirrors the stages in the previous diagram. In the example
chain we first ingest the data via the `ImportFromLocalFile` class (an additional operation we added
via the [`operationsDeclarations.json`](./additional-config.md#operations-declarations)), which
streams the data from the CSV file into the next `GenerateElements` operation.

For the generator we have selected the built in `Neo4jCsvElementGenerator` class, this is already
set up to be able to parse a correctly formatted neo4j exported CSV into Gaffer elements via the
schema. If you are curious as to what the output of each operation is you can try run a subset of
this chain to see how the data changes on each one, the output should be returned back to you in the
server response section of the Swagger API.

## Querying Data

Once data is loaded in the graph its now possible to start querying the data to gain insight and
perform analytics. Querying in Gaffer can get fairly complex but generally simple queries are made

> up of two parts, a `Get` Operation and a `View`.

Reviewer 2: should be a semi-colon rather than a comma before 'a `Get`' :)
